Python/ceval.c

```diff
@@ -5199,6 +5192,17 @@ format_awaitable_error(PyTypeObject *type, int prevopcode)
     }
 }
 
+static inline void
+exit_thread_if_finalizing(PyThreadState *tstate)
```

Comment:
nitpick: you should move this function *before* PyEval_AcquireLock() to avoid the forward declaration. I prefer to declare functions before they are used :-)